src/codegen/formatted_elz/tests.rs

```diff
@@ -122,19 +123,102 @@ fn nested_block() {
 
 #[test]
 fn simple_trait() {
-    let program = parse_code("trait Foo{}");
+    let formatted_code = format_code("trait Foo{}");
     assert_eq!(
-        program.formatted_elz(DEFAULT_LEVEL),
+        formatted_code,
         "trait Foo {}
 "
     );
 }
 
+#[test]
+fn simple_comment() {
+    let formatted_code = format_code(
+        "//this is comment line
+trait Foo {}
+",
+    );
+    assert_eq!(
+        formatted_code,
+        "// this is comment line
+trait Foo {}
+"
+    )
+}
+
+#[test]
+fn multi_blank() {
+    let formatted_code = format_code("
+    class                 Car                               {name               :   string          ;new(name:string):Car;}
+    ");
+    assert_eq!(
+        formatted_code,
+        "class Car {
+  name: string;
+  new(name: string): Car;
+}
+"
+    );
+}
+
+#[test]
+fn multi_newline() {
+    let formatted_code = format_code(
+        "
+    class Car
+    {
+        name:string
+;
+new(name:string):Car
+;
+}
+",
+    );
+    assert_eq!(
+        formatted_code,
+        "class Car {
+  name: string;
+  new(name: string): Car;
+}
+"
+    );
+}
+
+#[test]
+fn test() {
+    let formatted_code = format_code(
+        "//this is comment line
+trait Foo {}
+    class Car{name:string       ;//comment line one
+//comment line two;
+    :  :     new(name:string):Car;   bar(i: int):void  ;  fn foo()   {b=555*10 ; a=1+b;}}class CarFoo{::bar(): void {return;}}",
+    );
+    assert_eq!(
+        formatted_code,
+        "// this is comment line
+trait Foo {}
+class Car {
+  name: string;
+  // comment line one
+  // comment line two;
+  ::new(name: string): Car;
+  bar(i: int): void;
+  fn foo() {
+    b = 555 * 10;
+    a = 1 + b;
+  }
+}
+class CarFoo {
+  ::bar(): void {
+    return;
+  }
+}
+"
+    );
+}
+
 // helpers, must put tests before this line
-fn parse_code(code: &'static str) -> FormatTopAstList {
-    let program = match Parser::parse_program("", code) {
-        Ok(program) => program,
-        Err(err) => panic!("{}", err),
-    };
-    FormatTopAstList(program)
+fn format_code(code: &str) -> String {
+    let s = formatted_elz(code.to_string());
```

Comment:
I don't think we need to wrap ”formatted_elz.”

Directly call it would be better.

And why not name it ”format_code”? Notice the original name is because it belongs to AST. And the mod also needs a new name, how about ”formatter”?